web/src/stream_popover.js

```diff
@@ -515,14 +515,22 @@ export function register_click_handlers() {
         e.stopPropagation();
     });
 
-    $("#streamlist-toggle-button").on("click", (e) => {
-        e.preventDefault();
-        e.stopPropagation();
-
-        const sidebarHidden = !$(".app-main .column-left").hasClass("expanded");
-        popovers.hide_all();
-        if (sidebarHidden) {
-            show_streamlist_sidebar();
-        }
-    });
+    document.addEventListener(
+        "click",
+        (e) => {
+            const $left_column = $(".app-main .column-left");
+            const sidebarHidden = !$left_column.hasClass("expanded");
+            const $elt = $(e.target);
+            const toggle_button_clicked = Boolean($elt.closest("#streamlist-toggle-button").length);
+            const click_outside_left_sidebar = !$elt.closest($left_column).length;
+            if (!sidebarHidden && (click_outside_left_sidebar || toggle_button_clicked)) {
+                hide_streamlist_sidebar();
+            }
+
+            if (sidebarHidden && toggle_button_clicked) {
+                show_streamlist_sidebar();
+            }
+        },
+        {capture: true},
+    );
```

Comment:
It feels a bit bad to have this duplicate the `right_sidebar_ui.js` copy without any clear relationship between them. 

Could argue for renaming `right_sidebar_ui.js` to `sidebars_ui.js` to have them together there.

Reply:
Opened https://chat.zulip.org/#narrow/stream/6-frontend/topic/Introduce.20sidebars_ui.2Ejs.3F/near/1648356 to discuss this.